.github/workflows/helm-check.yaml

```diff
@@ -30,51 +30,56 @@ jobs:
           cd install/kubernetes
           make all
           git diff --exit-code
-  lint-test:
+
+  lint:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout
-        uses: actions/checkout@v1
+        uses: actions/checkout@v2
 
       - name: Run helm lint
         run: |
           cd install/kubernetes/cilium
           helm lint --with-subcharts --values values.yaml
 
+  conformance-test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
       - name: Build docker images
         run: |
           make docker-image-no-clean
           make docker-operator-generic-image
 
-      - name: Prevent K8s from pulling images
-        run: |
-          sed -i 's;pullPolicy: Always;pullPolicy: Never;g' install/kubernetes/cilium/values.yaml
-
       - name: Create kind cluster
         uses: helm/kind-action@v1.0.0-rc.1
         with:
           version: v0.8.1
+          config: .github/kind-config.yaml
 
       - name: Load local images into kind cluster
         run: |
           kind load docker-image --name chart-testing cilium/cilium:latest
           kind load docker-image --name chart-testing cilium/operator-generic:latest
 
-      # This test is just to make sure the helm chart can be installed. It will not replace existing integration tests
-      - name: Run chart-testing (install)
-        uses: helm/chart-testing-action@v1.0.0-rc.2
-        with:
-          command: install
-          config: install/kubernetes/chart-testing.yaml
-
-      - name: Install ginkgo
+      - name: Install cilium chart
         run: |
-          cd $HOME
-          go get -u github.com/onsi/ginkgo/ginkgo
-          go get -u github.com/onsi/gomega/...
+          helm install cilium ./install/kubernetes/cilium \
+             --wait \
+             --namespace kube-system \
+             --set global.nodeinit.enabled=true \
+             --set global.kubeProxyReplacement=partial \
+             --set global.hostServices.enabled=false \
+             --set global.externalIPs.enabled=true \
+             --set global.nodePort.enabled=true \
+             --set global.hostPort.enabled=true \
+             --set global.pullPolicy=Never
 
-      - name: Run test suite
+      - name: Run conformance test (e.g. connectivity check)
+        env:
+          CONFORMANCE_TEMPLATE: examples/kubernetes/connectivity-check/connectivity-check.yaml
         run: |
-          cd test
-          export PATH=$PATH:$HOME/go/bin
-          ginkgo -v --focus="K8sConformance Portmap Chaining Check one node" -- -cilium.provision=false -test.v --cilium.kubeconfig=${HOME}/.kube/config -cilium.image=cilium/cilium:latest -cilium.operator-image=cilium/operator-generic:latest
```

Comment:
Any reason the `K8sConformance` test isn't executed anymore?

Reply:
I think the existing conformance test is to perform connectivity check for _single node_ . This changes are to extend it with multiple nodes, some syntactic sugar with helm + kubectl apply to reduce dependency download time.

Based on the recent run, it is taking 10 mins (5 mins for docker build, and 3 mins for cilium installation + connectivity check).

Let me know what you have in mind :).

Reply:
You're right, it's running https://github.com/cilium/cilium/blob/819185568eb5cf653052043c5299d1224460368b/test/k8sT/Conformance.go#L83
I think you want to run this instead: https://github.com/cilium/cilium/blob/819185568eb5cf653052043c5299d1224460368b/test/k8sT/Conformance.go#L69

So the following should do it:
```diff
-           ginkgo -v --focus="K8sConformance Portmap Chaining Check one node" -- -cilium.provision=false -test.v --cilium.kubeconfig=${HOME}/.kube/config -cilium.image=cilium/cilium:latest -cilium.operator-image=cilium/operator-generic:latest
+           ginkgo -v --focus="K8sConformance Portmap Chaining Check connectivity-check" -- -cilium.provision=false -test.v --cilium.kubeconfig=${HOME}/.kube/config -cilium.image=cilium/cilium:latest -cilium.operator-image=cilium/operator-generic:latest
```

Reply:
Discussed offline. I misunderstood this initially.
It doesn't run ginkgo on purpose and just does a simple connectivity check. Given that the smoke test now takes 10min where it was taking 15min before, I'm :+1: with this if @cilium/ci folks are.